### PR TITLE
SAMZA-1202: Multiple calls to getInputStream() result in non-deterministic behavior

### PR DESCRIPTION
**Pull request by vjagadish1989**

### CODE REVIEW

**Comment by vjagadish1989** [comment]
cc: @jmakes @prateekm @xinyuiscool . Can you guys take a look?

File: samza-api/src/main/java/org/apache/samza/operators/StreamGraph.java [hunk 1]
```diff
@@ -39,6 +39,7 @@
    * @param <V> the type of message in the incoming message
    * @param <M> the type of message in the input {@link MessageStream}
    * @return the input {@link MessageStream}
+   * @throws IllegalStateException when invoked multiple times with the same {@code streamId}
```
Comment (prateekm) [inline samza-api/src/main/java/org/apache/samza/operators/StreamGraph.java]: Should document in description as well.

File: samza-api/src/main/java/org/apache/samza/system/StreamSpec.java [hunk 2]
```diff
@@ -201,4 +201,18 @@ private void validateLogicalIdentifier(String identifierName, String identifierV
       throw new IllegalArgumentException(String.format("Identifier '%s' is '%s'. It must match the expression [A-Za-z0-9_-]+", identifierName, identifierValue));
     }
   }
+  @Override
```
Comment (prateekm) [inline samza-api/src/main/java/org/apache/samza/system/StreamSpec.java]: Newline before.

File: samza-core/src/main/java/org/apache/samza/operators/StreamGraphImpl.java [hunk 3]
```diff
@@ -65,6 +65,11 @@ public StreamGraphImpl(ApplicationRunner runner, Config config) {
     if (msgBuilder == null) {
       throw new IllegalArgumentException("msgBuilder can't be null for an input stream");
     }
+
+    if (inStreams.containsKey(runner.getStreamSpec(streamId))) {
```
Comment (prateekm) [inline samza-core/src/main/java/org/apache/samza/operators/StreamGraphImpl.java]: Should we add this to getOutputStream as well?

**Comment by vjagadish1989** [comment]
Address @prateekm 's review feedback.

**Pull request closed without merging**